independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ContextInstancesGenerator.java

```diff
@@ -92,11 +93,20 @@ Collection<Resource> generate(DotName scope) {
         implementGetIfPresent(contextInstances, beans, idToFields);
         implementRemove(contextInstances, beans, idToFields);
         implementGetAllPresent(contextInstances, idToFields);
-        implementRemoveEach(contextInstances, idToFields);
+        // we have to choose between two implementations of removeEach:
+        // - one with a bulk lock/unlock for all fields
+        // - one with an unguarded check for null and lock/unlock for each field
+        // The first one has a smaller stack map table, while the second one has a faster path in case
+        // not all instances are being computed, yet.
+        if (idToFields.size() > 16) {
+            implementRemoveEach(contextInstances, idToFields);
 
-        // These methods are needed to significantly reduce the size of the stack map table for getAllPresent() and removeEach()
-        implementLockAll(contextInstances, idToFields);
-        implementUnlockAll(contextInstances, idToFields);
+            // These methods are needed to significantly reduce the size of the stack map table for getAllPresent() and removeEach()
+            implementLockAll(contextInstances, idToFields);
+            implementUnlockAll(contextInstances, idToFields);
```

Comment:
The `lockAll()` and `unlockAll()` methods are also used in `getAllPresent()`, so either you need to rewrite `getAllPresent()` in a similar fashion, or you need to keep `lockAll()` and `unlockAll()`.

Reply:
thanks for checking, let me fix it, good point!